lib/src/parsing/file-parser.ts

```diff
@@ -198,3 +201,38 @@ function containsApiDeclaration(sourcePath: string): boolean {
   }
   return false;
 }
+
+/**
+ * Check for TypeScript errors
+ */
+function isValidTypeScript(sourcePath: string): boolean {
+  const program = ts.createProgram([sourcePath], {
+    target: ts.ScriptTarget.ESNext,
+    experimentalDecorators: true,
+    moduleResolution: ts.ModuleResolutionKind.NodeJs,
+    baseUrl: "./",
+    paths: {
+      "@airtasker/spot": ["./lib/src/lib"]
+    }
+  });
+  let diagnostics = program.getSemanticDiagnostics();
```

Comment:
is this `getSemanticDiagnostics()` enough? Do we need to do `getSyntheticDiagnostics()`?